test/regression/issue/02499-repro.ts

```diff
@@ -19,3 +19,5 @@ console.log(
     }
   }
 })();
+
+export {};
```

Comment:
is there a way to tell typescript that this isn't necessary?

Reply:
I tried setting `"type": "module"` in test/package.json which should theoretically work, but it didn't. It might be a bug with TS or VSCode, or something else I'm missing. Adding `export {}` and renaming the file to `*.mts` both work.

Reply:
Ah. It's because we are using `"moduleResolution": "bundler"`. Currently TS switched to treating *.ts as a module only when type=module and moduleResolution is `node16`+

Reply:
@andrewbranch do you know if there's an existing way to tell TypeScript to assume every file is a module & allow optional file extensions (also allowing .ts extensions to be imported)? 

The original issue was the following two different entry points would cause a TS error because both declare a top-level block-scoped variable:

```ts
// test-repro-a.ts
const server = Bun.serve({
  port: 0,
  async fetch(req) {
    console.log(await req.json());
    return new Response();
  },
});
```

```ts
// test-repro-b.ts
const server = Bun.serve({
  port: 0,
  async fetch(req) {
    console.log(await req.json());
    return new Response();
  },
});
```

It sounds like TS is considering these as one top-level namespace, but they are different entry points

Reply:
`"moduleDetection": "force"` and `"allowImportingTsExtensions": true`. I would probably encourage Bun to ship a tsconfig.json with the correct settings that users can extend:

```json5
{
  "extends": "bun-types",
  // ...
}
```

While these settings make sense for basically all bundlers, `moduleResolution` is strictly speaking the wrong place to imply them as defaults. I’m considering a new `module` setting in the future that could wrap up all these defaults together.